hail/src/test/scala/is/hail/expr/ir/Aggregators2Suite.scala

```diff
@@ -453,4 +453,15 @@ class Aggregators2Suite extends HailSuite {
 
     assertAggEquals(grouped2, initOpArgs, seqOpArgs, expected = expected, args = FastIndexedSeq(("rows", (arrayType, rows))))
   }
+
+  @Test def testCollectAsSet() {
+    val rows = FastIndexedSeq(Row("abcd", 5L), null, Row(null, -2L), Row("abcd", 7L), null, Row("foo", null))
+    val rref = Ref("rows", TArray(t))
+    val elts = Array.tabulate(rows.length)(i => FastIndexedSeq(GetField(ArrayRef(rref, i), "a")))
+
+    val expected = Set("abcd", "foo", null)
+
+    val aggsig = AggSignature2(CollectAsSet(), FastSeq(), FastSeq(TString()), None)
+    assertAggEquals(aggsig, FastSeq(), elts, expected = expected, args = FastIndexedSeq(("rows", (arrayType, rows))))
```

Comment:
can you add a test with no elements?

Reply:
er, actually add a test with a primitive key too, since those are handled differnetly.

Reply:
I added a test with no elements, but all that is in this case is checking that the result from a state that has had the init op called but no seq ops returned as expected, so I did that by adding an optional parameter on `assertAggEquals` to check the expected result of the initialized state.